cpp/inc/bond/comm/transport/epoxy.h

```diff
@@ -944,81 +913,133 @@ class EpoxyTransport
-                            boost::make_shared<EpoxyConnection>(m_allocator,
-                                                                address,
-                                                                std::move(this->m_socket));
+                boost::shared_ptr<EpoxyConnection> connection =
+                    boost::make_shared<EpoxyConnection>(m_allocator,
+                                                        address,
+                                                        std::move(m_socket),
+                                                        m_ioService);
 
-                        boost::shared_ptr<INetworkConnectionSink> sink =
-                            m_handler->ConnectionAccepted(connection,
-                                                          address,
-                                                          address);
-                        connection->Listen(sink);
+                boost::shared_ptr<INetworkConnectionSink> sink =
+                    m_handler->ConnectionAccepted(connection,
+                                                  address,
+                                                  address);
+                connection->Listen(sink);
 
-                        DoAccept();
-                    }
-                    else if (boost::asio::error::operation_aborted == ec)
-                    {
-                        BOND_LOG(LOG_INFO, "EpoxyServer",
-                                 "Accept aborted: " << ec.message());
-                    }
-                    else
-                    {
-                        BOND_LOG(LOG_ERROR, "EpoxyServer",
-                                 "Failed to accept: " << ec.message());
-                    }
-                });
-        }
+                DoAccept();
+            }
 
 
-        Allocator m_allocator;
+            Allocator m_allocator;
 
-        boost::asio::ip::tcp::acceptor m_acceptor;
+            boost::shared_ptr<boost::asio::io_service> m_ioService;
 
-        boost::asio::ip::tcp::socket m_socket;
+            boost::shared_ptr<INetworkServerSink> m_handler;
+
+            boost::asio::ip::tcp::socket m_socket;
+
+            boost::asio::ip::tcp::acceptor m_acceptor;
 
-        boost::shared_ptr<INetworkServerSink> m_handler;
+        public:
 
-        lockable<SinkMap> m_sinkMap;
+            EpoxyServerContext(const Allocator& allocator,
+                               const SocketAddress& address,
+                               const boost::shared_ptr<INetworkServerSink>& handler,
+                               const boost::shared_ptr<boost::asio::io_service>& ioService)
+                : m_allocator(allocator)
+                , m_ioService(ioService)
+                , m_handler(handler)
+                , m_socket(*ioService)
+                , m_acceptor(*ioService)
+            {
+                BOOST_ASSERT(nullptr != m_handler);
+                BOOST_ASSERT(nullptr != m_ioService);
```

Comment:
In reference to my other comment about unconditionally dereferencing empty shared_ptr instances, we appear to be asserting here, but the asserts will run after the dereference.